pkg/codegen/docs/gen_creation_example_for_yaml.go

```diff
@@ -0,0 +1,193 @@
+package docs
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+)
+
+// collapseToken converts an exact token to a token more suitable for
+// display. For example, it converts
+//
+//	  fizz:index/buzz:Buzz => fizz:Buzz
+//	  fizz:mode/buzz:Buzz  => fizz:mode:Buzz
+//		 foo:index:Bar	      => foo:Bar
+//		 foo::Bar             => foo:Bar
+//		 fizz:mod:buzz        => fizz:mod:buzz
+func collapseResourceTokenForYAML(token string) string {
+	tokenParts := strings.Split(token, ":")
+
+	if len(tokenParts) == 3 {
+		title := func(s string) string {
+			r := []rune(s)
+			if len(r) == 0 {
+				return ""
+			}
+			return strings.ToTitle(string(r[0])) + string(r[1:])
+		}
+		if mod := strings.Split(tokenParts[1], "/"); len(mod) == 2 && title(mod[1]) == tokenParts[2] {
+			// aws:s3/bucket:Bucket => aws:s3:Bucket
+			// We recourse to handle the case foo:index/bar:Bar => foo:index:Bar
+			tokenParts = []string{tokenParts[0], mod[0], tokenParts[2]}
+		}
+
+		if tokenParts[1] == "index" || tokenParts[1] == "" {
+			// foo:index:Bar => foo:Bar
+			// or
+			// foo::Bar => foo:Bar
+			tokenParts = []string{tokenParts[0], tokenParts[2]}
+		}
+	}
+
+	return strings.Join(tokenParts, ":")
+}
+
+func genCreationExampleSyntaxYAML(r *schema.Resource) string {
+	indentSize := 0
+	buffer := bytes.Buffer{}
+	write := func(format string, args ...interface{}) {
+		buffer.WriteString(fmt.Sprintf(format, args...))
+	}
+
+	indent := func() {
+		buffer.WriteString(strings.Repeat(" ", indentSize))
+	}
+
+	indended := func(f func()) {
+		indentSize += 2
+		f()
+		indentSize -= 2
+	}
+
+	seenTypes := codegen.NewStringSet()
+	var writeValue func(valueType schema.Type)
+	writeValue = func(valueType schema.Type) {
+		switch valueType {
+		case schema.BoolType:
+			write("True|False")
+		case schema.IntType:
+			write("0")
+		case schema.NumberType:
+			write("0.0")
+		case schema.StringType:
+			write("\"string\"")
```

Comment:
Normally strings aren't quoted in YAML.  Technically this doesn't need to be and probably shouldn't be here.  But I'm slightly torn as it does make it a little more clear where the "input" values are.  Would love to see more complex examples to decide whether the quotes are a good idea or not.